src/azure-cli/azure/cli/command_modules/iot/custom.py

```diff
@@ -109,38 +110,50 @@ def iot_dps_create(cmd, client, dps_name, resource_group_name, location=None, sk
     return client.iot_dps_resource.begin_create_or_update(resource_group_name, dps_name, dps_description)
 
 
-def iot_dps_update(client, dps_name, parameters, resource_group_name, tags=None):
+def iot_dps_update(client, dps_name, parameters, resource_group_name=None, tags=None):
+    resource_group_name = _ensure_dps_resource_group_name(client, resource_group_name, dps_name)
     if tags is not None:
         parameters.tags = tags
     return client.iot_dps_resource.begin_create_or_update(resource_group_name, dps_name, parameters)
 
 
-def iot_dps_delete(client, dps_name, resource_group_name):
+def iot_dps_delete(client, dps_name, resource_group_name=None):
+    resource_group_name = _ensure_dps_resource_group_name(client, resource_group_name, dps_name)
     return client.iot_dps_resource.begin_delete(dps_name, resource_group_name)
 
 
-# DPS access policy methods
-def iot_dps_access_policy_list(client, dps_name, resource_group_name):
-    iot_dps_get(client, dps_name, resource_group_name)
+# DPS policy methods
+def iot_dps_policy_list(client, dps_name, resource_group_name=None):
+    resource_group_name = _ensure_dps_resource_group_name(client, resource_group_name, dps_name)
     return client.iot_dps_resource.list_keys(dps_name, resource_group_name)
 
 
-def iot_dps_access_policy_get(client, dps_name, resource_group_name, access_policy_name):
-    iot_dps_get(client, dps_name, resource_group_name)
+def iot_dps_policy_get(client, dps_name, access_policy_name, resource_group_name=None):
+    resource_group_name = _ensure_dps_resource_group_name(client, resource_group_name, dps_name)
     return client.iot_dps_resource.list_keys_for_key_name(dps_name, access_policy_name, resource_group_name)
 
 
-def iot_dps_access_policy_create(cmd, client, dps_name, resource_group_name, access_policy_name, rights, primary_key=None, secondary_key=None, no_wait=False):
+def iot_dps_policy_create(
+    cmd,
+    client,
+    dps_name,
+    access_policy_name,
+    rights,
+    resource_group_name=None,
+    primary_key=None,
+    secondary_key=None,
+    no_wait=False
+):
+    resource_group_name = _ensure_dps_resource_group_name(client, resource_group_name, dps_name)
     dps_access_policies = []
-    dps_access_policies.extend(iot_dps_access_policy_list(client, dps_name, resource_group_name))
+    dps_access_policies.extend(iot_dps_policy_list(client, dps_name, resource_group_name))
     if _is_policy_existed(dps_access_policies, access_policy_name):
         raise CLIError("Access policy {0} already existed.".format(access_policy_name))
```

Comment:
`Access policy {0} already exists.` ?

Reply:
It is a way to format strings - you can specify numbers/variables (and have them repeat if needed)

Reply:
That wasn't the point of my comment compare the grammar.